src/components/textarea/bl-textarea.ts

```diff
@@ -0,0 +1,277 @@
+   */
+  @property({ type: String, reflect: true })
+  size?: TextareaSize = 'medium';
+
+  /**
+   * Sets label of the textarea
+   */
+  @property({})
+  label?: string;
+
+  /**
+   * Makes label as fixed positioned
+   */
+  @property({ type: Boolean, attribute: 'label-fixed' })
+  labelFixed = false;
+
+  /**
+   * Sets placeholder of the textarea
+   */
+  @property({})
+  placeholder?: string;
+
+  /**
+   * Enables showing character counter.
+   */
+  @property({ type: Boolean, attribute: 'character-counter' })
+  characterCounter = false;
+
+  /**
+   * Adds help text
+   */
+  @property({ type: String, attribute: 'help-text' })
+  helpText?: string;
+
+  /**
+   * Set custom error message
+   */
+  @property({ type: String, attribute: 'invalid-text' })
+  customInvalidText?: string;
+
+  /**
+   * Sets minimum length of the textarea
+   */
+  @property({ type: Number })
+  minlength?: number;
+
+  /**
+   * Sets max length of textarea
+   */
+  @property({ type: Number })
+  maxlength?:number;
+
+  /**
+   * Sets initial value of the textarea
+   */
+  @property()
+  value = '';
+
+  /**
+   * Sets textarea visible row count.
+   */
+  @property({ type: Number})
+  rows?: number = 4;
+
+  @event('bl-input') private onInput: EventDispatcher<string>;
+
+  @event('bl-change') private onChange: EventDispatcher<string>;
+
+  @event('bl-invalid') private onInvalid: EventDispatcher<ValidityState>;
+
+
+  private initialHeight:number;
+  private lineHeight:number;
+  private expandScroll = false;
+
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('invalid', this.onError);
+
+
+    this.internals.form?.addEventListener('submit', () => {
+      this.reportValidity();
+    });
+  }
+
+
+  private onError = (): void => {
+    this.onInvalid(this.internals.validity);
+  }
+
+  private  inputHandler(event: Event) {
+    if(this.expand){
+      this.autoResize();
+    }
+
+
+    const value = (event.target as HTMLTextAreaElement).value;
+
+    this.setValue(value);
+    this.onInput(value);
+  }
+
+  private changeHandler(event: Event) {
+    const value = (event.target as HTMLTextAreaElement).value;
+
+    this.dirty = true;
+    this.setValue(value);
+    this.onChange(value);
+  }
+
+  firstUpdated() {
+    this.setValue(this.value);
+  }
+
+  updateSlotted(event:Event) {
+    const _target = (event.target as HTMLSlotElement)
+    const value = _target.assignedNodes().map((n) => n.textContent).join('');
+    this.value = value
+    this.setValue(this.value);
+  }
+
+  reportValidity() {
+    this.dirty = true;
+    return this.checkValidity();
+  }
+
+  valueChangedCallback(value: string): void {
+    this.value = value;
+  }
+
+  validityCallback(): string | void {
+    return this.customInvalidText || this.validationTarget?.validationMessage;
+  }
+
+  private autoResize() {
+    const verticalPadding = parseInt(getComputedStyle(this.validationTarget).padding[0]) * 2;
```

Comment:
`getComputedStyle` is an expensive operation and we do this in every keypress when expand mode is enabled. We may look for an option that we can ignore padding sizes (like using `box-sizing: content-box` in css).

Reply:
if I only call getComputedStyle when component first created( I mean in connectedCallback), is it still expensive for our logic?

Reply:
I you do that calculation in `connectedCallback` that will mean it'll not work properly if that value changes after component rendered (maybe with a CSS media query). So I would not prefer that.

Reply:
I assigned verticalPadding in firstUpdated() method. So, it called only once when component first updated.